controller/server/gui.ml

```diff
@@ -196,9 +196,14 @@ module NetworkGui = struct
 
     let proxy = Proxy.from_default_service all_services in
 
+    let check_timeout =
+      try (Opium.Std.param req "timeout" |> float_of_string |> min 5.0 |> max 0.0)
+      with _ -> 0.2
+    in
```

Comment:
This is always returning `0.2` for me, it is not able to read the `timeout` query parameter.

`Opium.Std.param` seems to be reading from `Request.env`, which is empty:
https://github.com/rgrinberg/opium/blob/0.18.0/opium_kernel/router.ml#L38-L42

This question was asked to know of a better method to get query parameters:
`https://github.com/rgrinberg/opium/issues/32`

We currently don’t have `Request.query`, because we are on an older version of `opium`.

This code is working to get the `timeout`:

```ocaml
let check_timeout =
  let uri = Request.uri req in
  try (Uri.get_query_param uri "timeout"
    |> Option.get
    |> float_of_string
    |> min 5.0
    |> max 0.0
  ) with _ -> 0.2
in
```

Reply:
Sorry about my testing lapse and thanks for the fix!

Applied in d51e8f058c71e62f104266b4257bbb5f6bb2697a, used the opportunity to avoid exceptions altogether.